README.rst

```diff
@@ -1,203 +1,128 @@
 
-   EOX_HOOKS_ENROLLMENTS_BACKEND: "eox_hooks.edxapp_wrapper.backends.enrollments_l_v1"
-   EOX_HOOKS_COURSES_BACKEND: "eox_hooks.edxapp_wrapper.backends.courses_l_v1"
-   EOX_HOOKS_COURSE_MODES_BACKEND: "eox_hooks.edxapp_wrapper.backends.course_modes_l_v1"
-   EOX_HOOKS_MODELS_BACKEND: "eox_hooks.edxapp_wrapper.backends.models_l_v1"
-
-**Olive**
-
-.. code-block:: yaml
+    USE_EOX_HOOKS: True
+    EOX_HOOKS_DEFINITIONS:
+    {
+        "<trigger_event>":
+        {
+            "action": "<action_name>",
+            "config":
+            {
+                "url": "https://webhook.site",
+            },
+            "fail_silently": False,
+            "module": "<path_to_module>",
+        },
+    }
 
-   EOX_HOOKS_ENROLLMENTS_BACKEND: "eox_hooks.edxapp_wrapper.backends.enrollments_l_v1"
-   EOX_HOOKS_COURSES_BACKEND: "eox_hooks.edxapp_wrapper.backends.courses_l_v1"
-   EOX_HOOKS_COURSE_MODES_BACKEND: "eox_hooks.edxapp_wrapper.backends.course_modes_l_v1"
-   EOX_HOOKS_MODELS_BACKEND: "eox_hooks.edxapp_wrapper.backends.models_l_v1"
+Where:
+- **trigger_event:** The name of the event the platform sends and eox-hooks listen, such as ``post_enrollment``. For a detailed list, visit the `trigger events examples file <docs/trigger_event_examples.rst>`_.
+- **action_name:** Specify the name of the action to execute. For example: ``custom_action``.
+- **config:** Adds more information to the action.
+- **fail_silently:** Defines what happens if an exception arises while executing the action. If ``False``, then the exception is raised.
+- **path_to_module:** Path to the module defining the action to execute. For example: ``eox_hooks.actions``.
 
-**Palm and Quince**
+Example
+########
 
 .. code-block:: yaml
 
-   EOX_HOOKS_ENROLLMENTS_BACKEND: "eox_hooks.edxapp_wrapper.backends.enrollments_l_v1"
-   EOX_HOOKS_COURSES_BACKEND: "eox_hooks.edxapp_wrapper.backends.courses_p_v1"
-   EOX_HOOKS_COURSE_MODES_BACKEND: "eox_hooks.edxapp_wrapper.backends.course_modes_l_v1"
-   EOX_HOOKS_MODELS_BACKEND: "eox_hooks.edxapp_wrapper.backends.models_l_v1"
-
-
-These settings can be changed in ``eox_hooks/settings/common.py`` or, for example, in ansible configurations.
-
-**NOTE**:
-
-
-- The current ``common.py`` works with Open edX Lilac version.
-- Koa and Lilac work with PR (backports) 28266 and 28640 from Maple release.
-
-Open edX devstack
-*****************
-
-- Clone this repo in the src folder of your devstack.
-- Open a new Lms/Devstack shell.
-- Install the plugin as follows: pip install -e /path/to/your/src/folder
-- Restart Lms/Studio services.
-
-Usage
-#####
-
-Before using eox-hooks, this plugin must be configured using EOX_HOOKS_DEFINITIONS:
-
-.. code-block:: python
-
+    EOX_HOOKS_DEFINITIONS:
+    {
+        "post_register":
         {
-            "trigger_event": {
-                "module": "path_to_module",
-                "action": "action_name",
-                "fail_silently": True,
+            "action": "post_to_webhook_url",
```

Comment:
From what I understand the action can be custom and hosted on any path, right?

However, I think it would also be useful to add the actions that are already available in `eox-hooks`, similar to how we did with triggers, what do you think?

Reply:
I agree.

Reply:
I added the information, but please feel free to reopen the conversation if necessary.